ports/capnproto/vcpkg.json

```
@@ -1,10 +1,10 @@
 {
   "name": "capnproto",
   "version": "0.9.1",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Data interchange format and capability-based RPC system",
   "homepage": "https://capnproto.org/",
-  "supports": "!uwp & !((arm | arm64) & windows)",
+  "supports": "!uwp & (!windows | !arm)",
```

```suggestion
  "supports": "!windows | (!uwp & !arm)",
```

I like this better, since I think it's more clear what's happening - either you're on !windows, in which case it's all good, or you're on windows in which case you cannot be arm or uwp.

---

@strega-nil-ms wouldn't it be more clear if there was something like `"unsupportedPlatforms": "uwp | (windows & arm)"`?

---

@strega-nil-ms Yes, I'm happy to go all the way to DNF. Proof of equivalence:

```
!uwp & (!windows | !arm)          given
(!uwp & !windows) | (!uwp & !arm) distribute ands over ors
!windows | (!uwp & !arm)          !windows implies !uwp
```

@Osyotr "unsupported" is just "supports" with a ! wrapping the whole expression.

---

@BillyONeal yeah, but these negations are really confusing sometimes. Forcing DNF is also OK.